src/main/scala/com/fulcrumgenomics/bam/EstimatePoolingFractions.scala

```diff
@@ -0,0 +1,211 @@
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.bam
+
+import com.fulcrumgenomics.FgBioDef._
+import com.fulcrumgenomics.cmdline.{ClpGroups, FgBioTool}
+import com.fulcrumgenomics.util.{Io, Metric}
+import com.fulcrumgenomics.vcf.ByIntervalListVariantContextIterator
+import dagr.commons.util.LazyLogging
+import dagr.sopt.{arg, clp}
+import htsjdk.samtools.SamReaderFactory
+import htsjdk.samtools.util.SamLocusIterator.LocusInfo
+import htsjdk.samtools.util._
+import htsjdk.variant.variantcontext.VariantContext
+import htsjdk.variant.vcf.VCFFileReader
+import org.apache.commons.math3.stat.regression.OLSMultipleLinearRegression
+
+import scala.collection.JavaConversions.iterableAsScalaIterable
+
+@clp(group=ClpGroups.SamOrBam, description=
+  """
+    |Examines a pooled sample and estimates the fraction of each constituent sample.
+    |Uses a VCF of known genotypes for the samples within the mixture along with a
+    |BAM of sequencing data derived from the pool.  Performs a multiple regression
+    |for the minor allele fractions at each SNP locus, using as inputs the individual
+    |sample's genotypes.  Only SNPs that are bi-allelic within the pooled samples are
+    |used.
+    |
+    |Various filtering parameters can be used to control which loci are used:
+    | --intervals will restrict analysis to variants within the described intervals
+    | --min-genotype-quality will filter out any site with any genotype with GQ < n
+    | --min-coverage-per-sample requires that the coverage of a site in the BAM be
+    |     >= min-coverage-per-sample * n_samples
+    | --min-mapping-quality filters out reads in the BAM with MQ < n
+    | --min-base-quality filters out bases in the BAM with Q < n
+  """)
+class EstimatePoolingFractions
+( @arg(flag="v", doc="VCF of individual sample genotypes.")  val vcf: PathToVcf,
+  @arg(flag="b", doc="Path to BAM file of sequencing data.") val bam: PathToBam,
+  @arg(flag="o", doc="Output file to write with pooling fractions.") val output: FilePath,
+  @arg(flag="l", minElements=0, doc="Zero or more set of regions to restrict analysis to.") val intervals: Seq[PathToIntervals] = Seq.empty,
+  @arg(flag="s", minElements=0, doc="Optional subset of samples from VCF to use.") val samples: Seq[String] = Seq.empty,
+  @arg(flag="g", doc="Minimum genotype quality. Use -1 to disable.") val minGenotypeQuality: Int = 30,
+  @arg(flag="c", doc="Minimum sequencing coverage per sample.") val minCoveragePerSample: Int = 12,
+  @arg(flag="m", doc="Minimum mapping quality.") val minMappingQuality: Int = 30,
+  @arg(flag="q", doc="Minimum base quality.") val minBaseQuality:Int = 30
+) extends FgBioTool with LazyLogging {
+  Io.assertReadable(vcf :: bam :: intervals.toList)
+
+  /* Class to hold information about a single locus. */
+  case class Locus(chrom: String, pos: Int, ref: Char, alt: Char, fractions: Array[Double], var observedFraction: Option[Double] = None)
```

Comment:
could you name  fractions and observedFractions in a way that makes it clearer which is from the genotypes and which is from the reads?